code/datums/components/manual_breathing.dm

```diff
@@ -0,0 +1,97 @@
+/datum/component/manual_breathing
+	dupe_mode = COMPONENT_DUPE_UNIQUE
+
+	var/obj/item/organ/lungs/L
+	var/warn_grace = FALSE
+	var/warn_dying = FALSE
+	var/last_breath
+	var/check_every = 12 SECONDS
+	var/grace_period = 6 SECONDS
+	var/damage_rate = 1 // organ damage taken per tick
+	var/next_breath_type = /datum/emote/inhale
+
+/datum/component/manual_breathing/Initialize()
+	if(!iscarbon(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	var/mob/living/carbon/C = parent
+	L = C.getorganslot(ORGAN_SLOT_LUNGS)
+
+	if(L)
+		START_PROCESSING(SSdcs, src)
+		last_breath = world.time
+		to_chat(C, "<span class='notice'>You suddenly realize you're blinking manually.</span>")
+
+/datum/component/manual_breathing/Destroy(force, silent)
+	L = null
+	STOP_PROCESSING(SSdcs, src)
+	to_chat(parent, "<span class='notice'>You revert back to automatic blinking.</span>")
+	return ..()
+
+/datum/component/manual_breathing/RegisterWithParent()
+	RegisterSignal(parent, COMSIG_MOB_EMOTE, .proc/check_emote)
+	RegisterSignal(parent, COMSIG_CARBON_GAIN_ORGAN, .proc/check_added_organ)
+	RegisterSignal(parent, COMSIG_CARBON_LOSE_ORGAN, .proc/check_removed_organ)
+	RegisterSignal(parent, COMSIG_LIVING_REVIVE, .proc/restart)
+	RegisterSignal(parent, COMSIG_MOB_DEATH, .proc/pause)
+
+/datum/component/manual_breathing/UnregisterFromParent()
+	UnregisterSignal(parent, COMSIG_MOB_EMOTE)
+	UnregisterSignal(parent, COMSIG_CARBON_GAIN_ORGAN)
+	UnregisterSignal(parent, COMSIG_CARBON_LOSE_ORGAN)
+	UnregisterSignal(parent, COMSIG_LIVING_REVIVE)
+	UnregisterSignal(parent, COMSIG_MOB_DEATH)
+
+/datum/component/manual_breathing/proc/restart()
+	START_PROCESSING(SSdcs, src)
+
+/datum/component/manual_breathing/proc/pause()
+	STOP_PROCESSING(SSdcs, src)
+
+/datum/component/manual_breathing/process()
+	if(!L || !iscarbon(parent))
+		STOP_PROCESSING(SSdcs, src)
+		return
+
+	var/mob/living/carbon/C = parent
+
+	var/next_text = (next_breath_type == /datum/emote/inhale ? "inhale" : "exhale")
```

Comment:
If you typecast next_breath_type you can do `var/next_text = initial(next_breath_type.key)`